reservation/views.py

```diff
@@ -1 +1,40 @@
-from django.shortcuts import render
+import json
+from datetime               import date, datetime
+
+from django.views           import View
+from django.http            import HttpResponse, JsonResponse
+
+# from user.decorator         import login_check
+from reservation.models     import (
+    Reservation,
+    OrderStatus,
+    Review,
+)
+from user.models            import (
+    User,
+    WishList,
+    Host,
+)
+
+class ReservationView(View):
+    # @login_check
+    def get(self, request):
+        user = User.objects.prefetch_related("reservation").get(id=request.user.id)
+        current_date = datetime.today().strftime('%Y-%m-%d')
+        state = request.GET.get("tab", None)
+
+        reservation_states = {
+            "이전 예약": "check_out__lte",
+            "예정된 예약": "check_in__gte"
+        }
+        if state in reservation_states:
+            curr_state = reservation_states[state]
+            reservation_list = [{
+            "name" : reservation.stay.title,
+            "location" : reservation.stay.address,
+            "stay_date" : str(reservation.check_in.month) + "월 " + str(reservation.check_in.day) + "일 - "+ str(reservation.check_out.month) + "월 " + str(reservation.check_out.day) + "일",
+            }for reservation in user.reservation.filter(curr_state = current_date)]
+        
+        else:
+            return JsonResponse({"message": "invalid tab"}, status=404)
+        return JsonResponse({"stay_list": reservation_list}, status=200)
```

Comment:
예약 과정이 정상 작동하면 결과를 바로 표출하면 되기 때문에 200 return의 경우 위치는 if 문 내부가 더 적당합니다.

그런 다음 그 if문을 만족하지 못한다면 404가 나오므로 `else`가 필요치 않습니다.

```suggestion
            }for reservation in user.reservation.filter(curr_state = current_date)]
            
            return JsonResponse({"stay_list": reservation_list}, status=200)
            
        return JsonResponse({"message": "invalid tab"}, status=404)
```